### PR TITLE
doc: add environments.md

### PR DESCRIPTION
**Pull request by zr-msft**
**Description**: Added environments.md file to be inline with other open source repositories.

**Testing done**: Markdown file renders correctly and links work

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ x] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? NO
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change? NO

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? N/A

### CODE REVIEW

File: adopters.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Adopters
+
+This document lists the environments you can use OSM in as well as some additional OSM usage guidelines.
+
+## Environments
```
Comment (shashankram) [inline adopters.md:5]: Adopters and environments seem orthogonal to me.

It would make more sense to either:
1. rename this file to environments.md
2. Update the wording to document adoption of this project (e.g. https://github.com/projectcontour/contour/blob/main/ADOPTERS.md)

Comment (zr-msft) [reply adopters.md]: @shashankram updated PR to use environments.md since we don't have the content available to create adopters.md

File: environments.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Environments
+
+OSM can be installed in any Kubernetes cluster running in a variety of environments such as AWS, Azure Kubernetes Service, Google Cloud, and even bare metal.
+
+In the specific case of Azure Kubernetes Service, [a supported add-on](https://docs.microsoft.com/azure/aks/open-service-mesh-about) is provided to handle installation and configuration of OSM.
```
Comment (shashankram) [inline environments.md:5]: Should we add a disclaimer regarding the addon not being managed by the upstream OSS project but rather by Microsoft?

**Review by shashankram — APPROVED** [review]

**Comment by zr-msft** [comment]
@shashankram are there any additional reviewers needed for this PR? or are you willing to merge?

**Comment by shashankram** [comment]
> @shashankram are there any additional reviewers needed for this PR? or are you willing to merge?

@nojnhuh, @jaellio, @trstringer could one of you review and merge this?

**Review by nojnhuh — APPROVED** [review]
